gen_proto.sh

```diff
@@ -2,11 +2,11 @@
 
 set -eou pipefail
 
-namespace=pysubstrait
+namespace=proto
 submodule_dir=./third_party/substrait
 src_dir="$submodule_dir"/proto
-tmp_dir=./proto
-dest_dir=./src/substrait/proto
+tmp_dir=./buf_work_dir
+dest_dir=./src/substrait
```

Comment:
I think the hack-y way is to make this an intermediate dir and then rename it or put it in a different namespace (like `pysubstrait`) and then alias the imports via the `__init__.py` file?

Reply:
JK, the script just calls `protol`. Maybe remove the `--create-package` option and it will stop creating the empty `__init__.py` file

Reply:
ooh ill give this a try!

Reply:
It doesn't quite work as we'd like. --create-package will recursively create init files, which we do want (except at the top level). Changing --in-place to --not-in-place would disallow overwriting of files, which we don't quite want either. Ultimately the cleanest approach to me still feels like placing these files in a `proto` subdir.

Reply:
Unless I'm misunderstanding you, @gforsyth 's suggestion is a `proto` subdir. I just assumed that the `protol` tool is creating an extra `__init__.py` file at the same level as the `proto` subdir.

I imagine the directory layout to be something like:
```
substrait
    ├── __init__.py (protol generates this by default)
    ├── proto (this is the proto subdir)
        ├── algebra_pb2.py
        ├── plan_pb2.py
        ├── ...
```

Reply:
but that being said, I see you added a gen directory which is probably fine in the near-term. I'm not sure what's generating that `__init__.py` file

Reply:
Yep your initial understanding is correct! IMO it would be better to generate files into a subdir if `protol` doesn't support exactly what we want. I'm hesitant to introduce any opaque behavior for devs to reduce bugs in the future. Having to choose between overwriting the top level __init__.py or not recursively generating __init__.py's seems bad. I'd rather have both if it means generating files into a subdir instead of root dir.

Reply:
Oh and I forgot to mention that `protol` was *deleting* the top level __init__.py file when removing the `--create-package` option!

Reply:
That was because I had `--in-place` option set (aka overwrite).